Userland/Libraries/LibWeb/UIEvents/WheelEvent.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGUI/Event.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/EventNames.h>
+#include <LibWeb/UIEvents/EventNames.h>
+#include <LibWeb/UIEvents/WheelEvent.h>
+
+namespace Web::UIEvents {
+
+WheelEvent::WheelEvent(JS::Realm& realm, FlyString const& event_name, WheelEventInit const& event_init)
+    : MouseEvent(realm, event_name, event_init)
+    , m_delta_x(event_init.delta_x)
+    , m_delta_y(event_init.delta_y)
+    , m_delta_mode(event_init.delta_mode)
+{
+    set_prototype(&Bindings::cached_web_prototype(realm, "WheelEvent"));
+    set_event_characteristics();
+}
+
+WheelEvent::~WheelEvent() = default;
+
+WheelEvent* WheelEvent::create(JS::Realm& realm, FlyString const& event_name, WheelEventInit const& event_init)
+{
+    return realm.heap().allocate<WheelEvent>(realm, realm, event_name, event_init);
+}
+
+WheelEvent* WheelEvent::create_from_platform_event(JS::Realm& realm, FlyString const& event_name, double offset_x, double offset_y, double client_x, double client_y, double delta_x, double delta_y)
+{
+    WheelEventInit event_init {};
```

Comment:
This doesn't initialize any properties of the MouseEventInit. 

The spec steps here https://www.w3.org/TR/uievents/#event-type-wheel

have instructions for what each of the fields of UIEventInit and MouseEventInit should be on its WheelEventInit for a given WheelEvent.

Reply:
Ah I see. Looks like our MouseEvent is missing some of those properties too. I do wonder about the buttons property on the mouse event itself, but if that's all the fields we've got in the init struct then that'll be a refactor for another day.